pages/deployment/wifi.md

```diff
@@ -1,6 +1,60 @@
-# Wifi Guide
+# Network Setup
 
-## Connecting a Device to WiFi
+* [Introduction](/pages/deployment/wifi.md#introduction)
+* [Network Requirements](/pages/deployment/wifi.md#network-requirements)
+* [Ethernet Connections](/pages/deployment/wifi.md#ethernet-connections)
+* [Wifi Connections](/pages/deployment/wifi.md#wifi-connections)
+* [3G or Cellular Connections](/pages/deployment/wifi.md#3g-or-cellular-connections)
+* [Captive Portal Network Setup](/pages/deployment/wifi.md#captive-portal-network-setup)
+
+## Introduction
+
+In order to deploy code to your device(s), they will need to be connected to the internet in some shape or form. The resin.io HostOS manages your devices' network connection with [ConnMan][connman], an open source connection manager.
+
+Currently all resin.io devices are configured to favour [ethernet](/pages/deployment/wifi.md#ethernet-connections) and will automatically use this connection type if it is available. You can define all manner of network configurations using [Connman][connman], for example [setting a static IP](/pages/deployment/wifi.md#set-static-ip).
+
+## Network Requirements
+
+In order for a resin.io device to get outside of the local network and connect to the resin.io API, there are a few core network requirements.
+
+Resin.io makes use of the following ports:
+
+* `443` TCP - This is the most fundamental requirement - it is used to connect to the VPN and the web terminal, and many web endpoints using TLS (https://.)
+* `123` UDP - For NTP time synchronisation.
+* `53` UDP - For DNS name resolution.
+
+Each of these should work with outward only (and inward once outward connection established) firewall settings.
+
+Additionally, if the network your device is connecting to works with whitelisting, you should whitelist the following domains on Port `80` and `443`:
+* `*.resin.io`
+* `*.pubnub.com`
+* `*.ngrok.com`
+* `*.ngrok.io`
+
```

Comment:
We also make outgoing connection to mixpanel - though not a requirement - it will show up in their logs.